idaes/core/base/extended_control_volume1d.py

```diff
@@ -0,0 +1,123 @@
+            has_work_transfer: whether terms for work transfer should be
+                included in enthalpy balances
+            has_enthalpy_transfer: whether terms for enthalpy transfer due to
+                mass transfer should be included in enthalpy balance. This
+                should generally be the same as the has_mass_transfer
+                argument in the material balance methods
+            custom_term: a Python method which returns Pyomo expressions representing
+                custom terms to be included in enthalpy balances.
+                Method should accept time and phase list as arguments.
+
+        Returns:
+            Constraint object representing enthalpy balances
+        """
+        if has_heat_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_heat_transfer is False. "
+                "If you are trying to solve for heat duty to achieve isothermal operation, please use "
+                "a full energy balance as add a constraint to equate inlet and outlet temperatures."
+            )
+        if has_work_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_work_transfer is False. "
+                "If you are trying to solve for work under isothermal operation, please use "
+                "a full energy balance as add a constraint to equate inlet and outlet temperatures."
+            )
+        if has_enthalpy_transfer:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option does not support enthalpy transfer. "
+            )
+        if has_heat_of_reaction:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option requires that has_heat_of_reaction is False. "
+                "If you are trying to solve for heat duty to achieve isothermal operation, please use "
+                "a full energy balance as add a constraint to equate inlet and outlet temperatures."
+            )
+        if custom_term is not None:
+            raise ConfigurationError(
+                f"{self.name}: isothermal energy balance option does not support custom terms. "
+            )
+
+        # Add isothermal constraint
+        @self.Constraint(
+            self.flowsheet().time, self.length_domain, doc="Energy balances"
+        )
+        def enthalpy_balances(b, t, x):
+            if (
+                b.config.transformation_scheme != "FORWARD"
+                and x == b.length_domain.first()
+            ) or (
+                b.config.transformation_scheme == "FORWARD"
+                and x == b.length_domain.last()
+            ):
+                return Constraint.Skip
+
+            return (
+                b.properties[t, b.length_domain.prev(x)].temperature
+                == b.properties[t, x].temperature
```

Comment:
It looks like the constraint would be skipped at the outlet when using forward scheme, but what does this mean for temperature at the outlet?

Reply:
Actually, most of this is not needed as there is no partial derivative. The way this is written, it has to skip the first point no matter what.

Reply:
Fixed.

Reply:
Can we also rename `enthalpy_balances` to `isothermal_constraint` or something here?